### DISCUSSION TITLE
How to model _usage_current on wastewater structures with no link to the network

### DISCUSSION THREAD

**Discussion by ponceta** [QGEP/QGEP#834] [Q&A]
It is often problematic when dealing with private wastewater networks to show the correct usage of wastewater structures when their link to the network is sometimes unkown or doesn't exist :

![image](https://github.com/QGEP/QGEP/assets/8614223/fb7b13e3-8f79-43bd-9aa9-3733ef61d011)

My first idea is to create an attribute usage_expected on the wastewater_node to set manually the usage when the _usage_current is NULL or missing.

**Comment by ponceta** [comment]
Here is a suggestion on how to custom handle this :
https://github.com/VilleDePully/datamodel/commit/ff2b5cf016d14b0b0bf8e07acfb8e4a34fd1f584

But I would prefer to have it in core of the wastewater module.

**Comment by sjib** [comment]
You could use usage_planned

**Reply by ponceta** [reply]
`usage_planned` is at the channel level and is actually not inherited by wastewater_structures and it wouldn't work for unlinked wastewater stuctures.

**Reply by sjib** [reply]
You can inherit that to the structure easily from the outgoing channel. The VSA Wiki specifies usage_planned as `Utilisation planifiée par le concept d’évacuation des eaux` see https://vsa.ch/fr/wiki/2-1-6-indications-pour-genre_utilisation_actuel-et-genre_utilisation_prevu/

**Reply by sjib** [reply]
I see - your objects are not connected to a channel right now. Still the concept of usage_planned should fit as it defines what is allowed to connect there.

**Comment by urskaufmann** [comment]
I think it's the same color, as if the connected chanel has usage "unkown". But I think it's not generally defined, which color to use for usage "unkown". Because I use Grey for "Not in use", I normally use the same color as combined_wastewater...

**Reply by ponceta** [reply]
A specific color for unknown would be indeed a workaround until the right connexions are set.
